src/Adapter/CrudTrait.php

```diff
@@ -32,10 +35,10 @@ public function insert($table, array $data)
     {
         $table  = $this->quote()->identifier($table);
         $fields = implode(', ', array_keys($data));
```

Comment:
Is it worth improving to quote identifier the columns as well?

Reply:
Yeah, that would be good, as currently you have to manually quote `key`

Reply:
Are there any BC concerns with implementing that?

Reply:
This might be a BC break, as any implementation that's needing to manually add backticks would then be double-quoted.

Reply:
![image](https://user-images.githubusercontent.com/1084019/32794631-477a804a-c961-11e7-9c6a-32edfacd57f4.png)

Diff for this change. (Tests excluded for brevity)

Reply:
Because of BC, I'm leaning towards no change. We usually try to avoid protected words in our schemas (nowadays) but can't account for other implementations!